source/azure_iot_hub_client.c

```diff
@@ -583,6 +650,33 @@ void AzureIoTHubClient_Deinit( AzureIoTHubClient_t * pxAzureIoTHubClient )
 }
 /*-----------------------------------------------------------*/
 
+AzureIoTHubClientResult_t AzureIoTHubClient_SetSymmetricKey( AzureIoTHubClient_t * pxAzureIoTHubClient,
+                                                             const uint8_t * pucSymmetricKey,
+                                                             uint32_t ulSymmetricKeyLength,
+                                                             AzureIoTGetHMACFunc_t xHmacFunction )
+{
+    AzureIoTHubClientResult_t xResult;
+
+    if( ( pxAzureIoTHubClient == NULL ) ||
+        ( pucSymmetricKey == NULL ) || ( ulSymmetricKeyLength == 0 ) ||
+        ( xHmacFunction == NULL ) )
+    {
+        AZLogError( ( "AzureIoTHubClient set symmetric key failed: Invalid argument" ) );
+        xResult = eAzureIoTHubClientInvalidArgument;
+    }
+    else
+    {
+        pxAzureIoTHubClient->_internal.pucSymmetricKey = pucSymmetricKey;
+        pxAzureIoTHubClient->_internal.ulSymmetricKeyLength = ulSymmetricKeyLength;
+        pxAzureIoTHubClient->_internal.pxTokenRefresh = prvIoTHubClientGetToken;
```

Comment:
might be a dumb question but is there ever a scenario where someone wouldn't want the refresh done automatically?

Reply:
cert case, where you do not need SAS